### ISSUE TITLE
is missing in props validation ForwardRefRenderFunction

### ISSUE THREAD

**Issue by jonlepage** [yannickcr/eslint-plugin-react#3099]
Sry if am wrong , new with react, but, this should ok no!?

```tsx
type IfooProps = { e: string };
const Foo: ForwardRefRenderFunction<HTMLDivElement, IfooProps> = function Foo (props, ref) {
	const { e } = props;
	return	<div ref={ref}>hello</div>;
};
```
![image](https://user-images.githubusercontent.com/24865815/136094490-90cbed79-1167-47c6-b567-4d8599351565.png)

---

small letter is ok, but i think react not accept FC with small letter!
```tsx
type IfooProps = { e: string };
const foo: ForwardRefRenderFunction<HTMLDivElement, IfooProps> = function foo (props, ref) {
	const { e } = props;
	return	<div ref={ref}>hello</div>;
};
```
![image](https://user-images.githubusercontent.com/24865815/136094747-00bbabf0-66d0-4b86-9ead-738d148fe16e.png)

it seem props are not take by rule !

**Comment by vedadeepta** [comment]
is `ForwardRefRenderFunction` something that is imported from React?

**Comment by ljharb** [comment]
@lukasvice in your case, you're typing the variable, not the function - what if you do:
```js
export const Foo = function Foo(props, ref): FooRenderFunction {
  const { e } = props // <-- 'e' is missing in props validation
  return <div ref={ref}>hello</div>
}
```

**Comment by lukasvice** [comment]
Right, but the function arguments are correctly inferred by TypeScript, but not recognized by ESLint:

![image](https://user-images.githubusercontent.com/9607491/221616755-3cfe93a5-227c-42af-87be-fde136598de3.png)

In your example, the type `FooRenderFunction` is only the return type of the function and not the full function type.

**Comment by ljharb** [comment]
Then yes, please file a new issue, but i think one already exists - it’d be ideal to comment on that instead.
